src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
-    except niscope.Error as e:
-        assert e.code == -1074135027  # Error : Attribute is read-only.
-        assert e.description.find('Attribute is read-only.') != -1
-
-
-def test_acquisition_status(multi_instrument_session):
-    assert multi_instrument_session.acquisition_status() == niscope.AcquisitionStatus.COMPLETE
-
-
-def test_self_cal(multi_instrument_session):
-    multi_instrument_session.self_cal(niscope.Option.SELF_CALIBRATE_ALL_CHANNELS)
-
-
-def test_probe_compensation_signal(multi_instrument_session):
-    multi_instrument_session.probe_compensation_signal_start()
-    multi_instrument_session.probe_compensation_signal_stop()
-
-
-def test_configure_horizontal_timing(multi_instrument_session):
-    multi_instrument_session.configure_vertical(5.0, niscope.VerticalCoupling.DC)
-    multi_instrument_session.auto_setup()
-    multi_instrument_session.configure_horizontal_timing(10000000, 1000, 50.0, 1, True)
-    multi_instrument_session.trigger_modifier = niscope.TriggerModifier.AUTO
-    multi_instrument_session.configure_trigger_immediate()
-    multi_instrument_session.horz_record_length == 1000
-    multi_instrument_session.horz_sample_rate == 10000000
-
-
-def test_configure_chan_characteristics(multi_instrument_session):
-    multi_instrument_session.vertical_range = 4.0
-    multi_instrument_session.configure_chan_characteristics(50, 0)
-    assert 50.0 == multi_instrument_session.input_impedance
-
+    @pytest.fixture(scope='function')
+    def multi_instrument_session(self, grpc_options):
+        with niscope.Session(','.join(self._instruments), False, True, 'Simulate=1, DriverSetup=Model:5164; BoardType:PXIe', _grpc_options=grpc_options) as simulated_session:
+            yield simulated_session
 
-def test_filter_coefficients(session_5142):
-    assert [1.0] + [0.0] * 34 == session_5142.get_equalization_filter_coefficients()  # coefficients list should have 35 items
-    try:
-        filter_coefficients = [1.0, 0.0, 0.0]
-        session_5142.configure_equalization_filter_coefficients(filter_coefficients)
-    except niscope.Error as e:
-        assert "Incorrect number of filter coefficients." in e.description
-        assert e.code == -1074135024
-    filter_coefficients = [0.01] * 35
-    session_5142.configure_equalization_filter_coefficients(filter_coefficients)
-    assert filter_coefficients == session_5142.get_equalization_filter_coefficients()
-
-
-def test_send_software_trigger_edge(multi_instrument_session):
-    multi_instrument_session.send_software_trigger_edge(niscope.WhichTrigger.ARM_REFERENCE)
-
-
-def test_disable(multi_instrument_session):
-    assert multi_instrument_session.allow_more_records_than_memory is False
-    multi_instrument_session.allow_more_records_than_memory = True
-    multi_instrument_session.disable()
-    assert multi_instrument_session.allow_more_records_than_memory is False
-
-
-# Basic configuration tests
-def test_configure_ref_levels(single_instrument_session):
```

Comment:
`configure_ref_levels` isn't included in the .proto file, so two versions of `test_configure_ref_levels` exist. The original version was copied as is to `TestLibrary` while the new version in `TestGrpc` asserts that a `NotImplementedError` is raised.

Reply:
This is fine. I'm okay with not supporting this obsolete method right now. We can always change our mind and add it later. I like that we test for the NotImplementedError in TestGrpc.